### PR TITLE
Add new agnhost subcommand for a networking test

### PR DESCRIPTION
**Pull request by danwinship**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
One of the networking tests currently depends on parsing the output of wget to distinguish different errors. This adds a binary to agnhost that we can use instead. (Originally I had the agnhost binary exit with different statuses in the different error cases, but our test infrastructure makes it a lot easier to do string comparisons against the output than to check the exit status. Plus I didn't want the test to accidentally pass because `kubectl exec` happened to return one of the codes we were looking for on its own.

**Which issue(s) this PR fixes**:
None, but this is a spinoff of #79012 which in turn references https://github.com/kubernetes/kubernetes/issues/75324#issuecomment-489880420 and #76342.

```release-note
NONE
```

### CODE REVIEW

**Comment by danwinship** [comment]
/hold
assume the tests pass I'll want to break the iptables rules and ensure that that makes the test fail

File: test/images/agnhost/connect/connect.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2015 The Kubernetes Authors.
```
Comment (johnbelamaric) [inline test/images/agnhost/connect/connect.go]: nit: 2019

File: test/images/agnhost/connect/connect.go [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	"os"
+	"syscall"
+	"time"
+
+	"github.com/spf13/cobra"
+)
+
+// CmdConnect is used by agnhost Cobra.
+var CmdConnect = &cobra.Command{
+	Use:   "connect",
```
Comment (johnSchnake) [inline test/images/agnhost/connect/connect.go]: nit: the `Use` should show what the arg it expects is. Only the first word is used as the command name, the rest is for help messages.

```
Use:   "connect [tcp destination]",

**Comment by danwinship** [comment]
oh, silly me, I was confusing `e2epod.NewExecPodSpec` with `e2epod.newExecPodSpec`

**Comment by danwinship** [comment]
Ah, and it doesn't use a freshly-built agnhost anyway, it uses `gcr.io/kubernetes-e2e-test-images/agnhost:2.1`.

Right, ok, so I need to build my own agnhost image, run the test locally to confirm that it works, then I guess drop the `test/e2e/network` part of this PR and make it just a PR to add `agnhost connect`, get it merged, get agnhost rebuilt, and then we can update #79012 ?

**Comment by danwinship** [comment]
/hold cancel
tested locally, but this now only contains the agnhost change (and bumps `test/image/agnhost/VERSION`) and then we can get that rebuilt so the test can depend on it.

File: test/images/agnhost/VERSION [hunk 3]
```diff
@@ -1 +1 @@
-2.1
+2.2
```
Comment (spiffxp) [inline test/images/agnhost/VERSION]: Oh the joy of serializing PR's that bump this image.  I have already built/pushed https://github.com/kubernetes/kubernetes/pull/79142 which is 2.2, and  https://github.com/kubernetes/kubernetes/pull/76838 is pending as 2.3, can you bump this to 2.4?

**Comment by danwinship** [comment]
> can you bump this to 2.4?

done, but I'm just going to get needs-rebased anyway won't I? Maybe it would be easier to have the individual PRs not change it and just have whoever does the actual image builds push a trivial PR changing the version number only right before the build?

**Comment by danwinship** [comment]
Although I guess it's going to conflict with README.md, etc, as well too so there'd be conflicts anyway.

(not sure what just happened with the tests... looks like temporary infrastructure problems?)

**Comment by danwinship** [comment]
since this is ready and #76838 isn't would it make sense to repush this with `VERSION` 2.3 and let the other one be 2.4 when it's ready?

**Comment by spiffxp** [comment]
@danwinship FWIW I pushed `gcr.io/agnhost:2.4` from this PR, but I'm fine proceeding as you suggest if @bclau is fine with it

**Comment by spiffxp** [comment]
https://gist.github.com/spiffxp/700e6d6d7eb45a4dc62b268efe55681e

**Comment by bclau** [comment]
nice, thanks for adding this, this will make things easier in the future. :)

I'm fine with merging this first. I'll rebase my PRs afterwards.

**Comment by danwinship** [comment]
OK... @spiffxp given that you already pushed a 2.4 build, want to just merge this PR as-is, skipping over version 2.3, and then the windows fixes can become 2.5?

**Pull request merged**